runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -853,6 +853,14 @@ TR_J9ServerVM::getObjectClassAt(uintptr_t objectAddress)
    return std::get<0>(stream->read<TR_OpaqueClassBlock *>());
    }
 
+TR_OpaqueClassBlock *
+TR_J9ServerVM::getObjectClassFromKnownObjectIndex(TR::Compilation *comp, TR::KnownObjectTable::Index idx)
+   {
+   JITServer::ServerStream *stream = _compInfoPT->getMethodBeingCompiled()->_stream;
+   stream->write(JITServer::MessageType::VM_getObjectClassFromKnownObjectIndex, idx);
+   return std::get<0>(stream->read<TR_OpaqueClassBlock *>());
+   }
+
```

Comment:
At an index we have an object of a particular class and we could cache that relationship until a class redefinition happens that changes that relationship. This would need a reverse lookup to find the class that is redefined. If these messages are not that many we could leave with them.